src/components/BottomMenu.jsx

```diff
@@ -0,0 +1,92 @@
+import React, { useState } from "react";
+import { SafeAreaView, StyleSheet, TouchableOpacity, View } from "react-native";
+import HomeIcon from "./Home";
+import MentalIcon from "./Mental";
+import GrowIcon from "./Grow";
+import SettingsIcon from "./Settings";
+
+const BottomMenu = () => {
+  const [selectedButton, setSelectedButton] = useState(null);
+
+  const handlePress = (buttonIndex) => {
+    setSelectedButton(buttonIndex);
+  };
+
+  const renderIcon = (buttonIndex, isSelected) => {
+    const fill = isSelected ? "white" : "#8E8E8E";
+    switch (buttonIndex) {
+      case 0:
+        return <HomeIcon fill={fill} width={24} height={24} />;
+      case 1:
+        return <MentalIcon fill={fill} width={24} height={24} />;
+      case 2:
+        return <GrowIcon fill={fill} width={24} height={24} />;
+      case 3:
+        return <SettingsIcon fill={fill} width={24} height={24} />;
+      default:
+        return null;
+    }
+  };
+
+  return (
+    <SafeAreaView style={styles.BottomMenuContainer}>
+      <View style={styles.menuBar}>
+        <View style={styles.buttonsContainer}>
+          {[0, 1, 2, 3].map((buttonIndex) => (
+            <TouchableOpacity
+              key={buttonIndex}
+              style={[
+                styles.circleButton,
+                selectedButton === buttonIndex && styles.selectedCircleButton,
+              ]}
+              onPress={() => handlePress(buttonIndex)}
+            >
+              {renderIcon(buttonIndex, selectedButton === buttonIndex)}
+            </TouchableOpacity>
+          ))}
+        </View>
+      </View>
+    </SafeAreaView>
```

Comment:
There is not much point in using map and switch just to implement tracking indexes.

You can simplify this code by using all 4 buttons separately, so without map and switch, to implement color switching behaviour, create a basic "switchable button" that switches its state on click, and then render each of these buttons with customized content.